app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,41 @@
+class Admin::UsersController < AdminController
+  before_action :logged_in_user, only: [:index, :update]
+  # before_action :correct_user, only: [:edit, :update] これ書いたままで Set As Adminをすると何も起きずにTopに戻ってしまう。
+
+  def index
+    @users = User.paginate(page: params[:page], per_page: 10)
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.admin == true
+      @user.admin = false
+      @user.save # @user.update(admin = false)
+    else
+      @user.admin = true
+      @user.save # @user.update(admin = true)
+    end
+    redirect_to admin_users_path
+  end  
```

Comment:
Please refactor this code...

You can say something like.

if @user.admin?
 @user.update(admin: false)
else
 @user.update(admin: true)
end